controllers/machinehealthcheck_controller_test.go

```diff
@@ -335,14 +330,15 @@ var _ = Describe("MachineHealthCheck Reconciler", func() {
 				for _, m := range rtc.expectNotRemediated() {
 					machine := &clusterv1.Machine{}
 					key := types.NamespacedName{Namespace: m.Namespace, Name: m.Name}
-					Expect(k8sClient.Get(ctx, key, machine)).To(Succeed())
+					Expect(testEnv.Get(ctx, key, machine)).To(Succeed())
 					Expect(machine.GetDeletionTimestamp().IsZero()).To(BeTrue())
 				}
 
 				for _, m := range rtc.expectRemediated() {
 					machine := &clusterv1.Machine{}
 					key := types.NamespacedName{Namespace: m.Namespace, Name: m.Name}
-					Expect(k8sClient.Get(ctx, key, machine)).ToNot(Succeed())
+					err := testEnv.Get(ctx, key, machine)
+					Expect(apierrors.IsNotFound(err)).To(BeTrue())
```

Comment:
Calling out that this is the only substantive change, I am not sure why remediated machines were not being deleted before in this test but now they error with `NotFound`. Maybe the sync interval was different and so reconciliation didn't happen?

Reply:
That's interesting, the new check looks better though